cmd/protoc-gen-go-grpc/go.mod

```diff
@@ -2,14 +2,16 @@ module google.golang.org/grpc/cmd/protoc-gen-go-grpc
 
 go 1.22
 
+replace google.golang.org/grpc => ../..
+
```

Comment:
`protoc-gen-go-grpc` is a separate binary from the gRPC Go  library. It is used to generate client and server stubs. Prior to https://github.com/grpc/grpc-go/pull/7438, `protoc-gen-go-grpc` didn't depend on gRPC at all. This PR added a test dependency on gRPC Go using a replace directive to specify current version of gRPC Go. 

After releasing `protoc-gen-go-grpc`, we discovered that using a replace directive in a binary module is not allowed (https://github.com/grpc/grpc-go/issues/7448). To resolve this, the replace directive was removed in https://github.com/grpc/grpc-go/pull/7451. Since then, we haven't updated the grpc-go version.

Updating the deps in this module may not be required.